torch/csrc/jit/codegen/cuda/scheduler/transpose.cpp

```diff
@@ -165,50 +274,73 @@ std::shared_ptr<TransposeParams> getTransposeHeuristics(
             std::vector<TensorView*> data{
                 domain_map.findReferenceFor(grouped_inputs_outputs[0]),
                 domain_map.findReferenceFor(grouped_inputs_outputs[1])};
             return std::make_unique<std::vector<TensorView*>>(std::move(data));
           });
-  auto& largest = largest_entry.get();
-  TORCH_INTERNAL_ASSERT(largest.size() == 2);
-  TensorView* largest1 = largest[0];
-  TensorView* largest2 = largest[1];
+  auto& reference_tensors = reference_tensors_entry.get();
+  TORCH_INTERNAL_ASSERT(reference_tensors.size() == 2);
+  TensorView* reference1 = reference_tensors[0];
+  TensorView* reference2 = reference_tensors[1];
   TORCH_INTERNAL_ASSERT(
-      largest1 != nullptr, "Unable to find reference tensor for group 1");
+      reference1 != nullptr, "Unable to find reference tensor for group 1");
   TORCH_INTERNAL_ASSERT(
-      largest2 != nullptr, "Unable to find reference tensor for group 2");
+      reference2 != nullptr, "Unable to find reference tensor for group 2");
 
   const int64_t device_multiprocessor_count =
       (int64_t)at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
 
-  int64_t max_input_dtype_size = 1;
-
-  size_t n_input_tensors = 0;
-  for (auto inp : ir_utils::filterByType<TensorView>(fusion->inputs())) {
-    max_input_dtype_size = std::max(
-        max_input_dtype_size,
-        (int64_t)dataTypeSize(inp->getDataType().value(), index_type));
-    n_input_tensors++;
-  }
-
-  auto ref_root = largest1->getMaybeRFactorDomain();
+  auto ref_root = reference1->getMaybeRFactorDomain();
+  std::vector<int64_t> shape;
+  shape.reserve(reference1->nDims());
   int64_t n_elems = 1;
   for (size_t ref_i = 0; ref_i < ref_root.size(); ref_i++) {
+    auto id = ref_root[ref_i];
+    auto concrete_id = domain_map.getComputeAtMap().getConcreteMappedID(
+        id, IdMappingMode::EXACT);
     auto inferred_val =
-        runtime_info.expressionEvaluator().evaluate(ref_root[ref_i]->extent());
+        runtime_info.expressionEvaluator().evaluate(concrete_id->extent());
     TORCH_INTERNAL_ASSERT(
         inferred_val.has_value(),
         "Error inferring size for pointwise scheduler: ",
         ref_root[ref_i]->extent()->toInlineString());
-    n_elems *= inferred_val.value().as<int64_t>();
+    int64_t size = inferred_val.value().as<int64_t>();
+    n_elems *= size;
+    shape.push_back(size);
   }
 
   auto params = std::make_shared<TransposeParams>("Transpose heuristics");
 
+  // If the problem size is small use small tile sizes.
+  if (n_elems < device_multiprocessor_count * kMaxTileSize * kMaxTileSize) {
+    params->tile_size1 = 8;
+    params->tile_size2 = 8;
+    // TODO: I was trying the following but I got silent wrong result
+    // params->tile_size1 = 8;
+    // params->tile_size2 = 4;
+    // This should not happen, because the correctness should be irrevalent to
+    // schedulers. We don't have to use tile size (8, 4), but we need to fix our
+    // bug in codegen.
```

Comment:
Is there an issue filed for this?

Reply:
Not yet. Not sure if I will have bandwidth to write a minimum repro, but at least I will file an issue with non-minimum repro after this PR gets merged.

Reply:
I can not reproduce this anymore :(